### PR TITLE
Refactoring of the containers

### LINKED ISSUE #406: Refactoring of the containers
The container file is getting bloated. I think it is time to divide it into different files and do some refactoring.

### PR DESCRIPTION
**Pull request by interesaaat**
Closes #406.

### CODE REVIEW

File: hummingbird/ml/operator_converters/_tree_commons.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def get_tree_implementation_by_config_or_depth(extra_config, max_depth, low=3, h
     """
     if constants.TREE_IMPLEMENTATION not in extra_config:
         if max_depth is not None and max_depth <= low:
-            return TreeImpl.gemm
+            return TreeImpl.perf_tree_trav
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/_tree_commons.py]: why this change?

**Review by ksaur — APPROVED** [review]

File: hummingbird/ml/operator_converters/_tree_commons.py (hunk 1 above)
Comment (interesaaat) [reply hummingbird/ml/operator_converters/_tree_commons.py]: Ok this is a typo from testing. Let me roll this back
